### PR TITLE
Fix: static file routing path rewrite.

### PR DESCRIPTION
**Pull request by fufuok**
**Please provide enough information so that others can review your pull request:**

Fixes: #1537

### CODE REVIEW

File: router.go [hunk 1]
```diff
@@ -327,8 +327,11 @@ func (app *App) registerStatic(prefix, root string, config ...Static) Router {
 			if len(path) >= prefixLen {
 				if isStar && app.getString(path[0:prefixLen]) == prefix {
 					path = append(path[0:0], '/')
-				} else if len(path) > 0 && path[len(path)-1] != '/' {
-					path = append(path[prefixLen:], '/')
+				} else {
+					path = path[prefixLen:]
+					if len(path) == 0 || path[len(path)-1] != '/' {
+						path = append(path, '/')
+					}
```
Comment (renanbastos93) [inline router.go:334]: Can you create unit test for it?

Comment (fufuok) [reply router.go:334]: Sure.

Comment (fufuok) [reply router.go:334]:
```
go test -v -run=^Test_Route_Static
=== RUN   Test_Route_Static_Root
--- PASS: Test_Route_Static_Root (0.01s)
=== RUN   Test_Route_Static_HasPrefix
--- PASS: Test_Route_Static_HasPrefix (0.00s)
PASS
ok      github.com/gofiber/fiber/v2     0.077s
```

**Review by renanbastos93 — APPROVED** [review]
Congrats, you made a good job.

**Review by ReneWerner87 — APPROVED** [review]

**Comment by ReneWerner87** [comment]
@fufuok please correct the imports

**Comment by fufuok** [comment]
> @fufuok please correct the imports

Done.

ioutil.TempFile instead of os.CreateTemp
ioutil.ReadAll instead of  io.ReadAll

Tested in go1.14 and go1.17.

```
# go version
go version go1.14.15 windows/amd64

# go test -v -run=^Test_Route_Static
=== RUN   Test_Route_Static_Root
--- PASS: Test_Route_Static_Root (0.22s)
=== RUN   Test_Route_Static_HasPrefix
--- PASS: Test_Route_Static_HasPrefix (0.01s)
PASS
ok      github.com/gofiber/fiber/v2     1.633s
```

```
# go version
go version go1.17 linux/amd64

# go test -v -run=^Test_Route_Static
=== RUN   Test_Route_Static_Root
--- PASS: Test_Route_Static_Root (0.02s)
=== RUN   Test_Route_Static_HasPrefix
--- PASS: Test_Route_Static_HasPrefix (0.01s)
PASS
ok      github.com/gofiber/fiber/v2     0.088s
```

**Comment by ReneWerner87** [comment]
@fufuok please check 
image

**Comment by fufuok** [comment]
https://github.com/gofiber/fiber/blob/master/app_test.go#L738
Why does the trailing slash here expect the result to be 404.

Please view this question gif and code: #1537

**Comment by ReneWerner87** [comment]
I think you are right, you can change the expectations

**Comment by fufuok** [comment]
image

Sorry, I rechecked the test case, because there is the default index.html in the directory, so it should be 200. I want to add another directory test case without index.html.

Unlike #1537 , it has option `Browse: true`

If possible, I would like to submit the above code, please approve it.

**Review by ReneWerner87 — APPROVED** [review]

**Pull request merged**